backends/google/sheets/google-sheets.js

```diff
@@ -138,8 +138,24 @@ export default class GoogleSheets extends Google {
 		}
 
 		if (this.options.headerRow === true || this.options.keys) {
+			// Reverse the objectKeys map
+			const columnNumbers = new Map([...file.objectKeys].map(([index, key]) => [key, index]));
+
 			// Transform an array of objects into an array of arrays.
-			data = data.map(obj => Object.values(obj));
+			// We must preserve the columns' source order.
+			data = data.map(obj => {
+				const ret = [];
+
+				for (const key of Object.keys(obj)) {
+					const index = columnNumbers.get(key) ?? key; // Why “?? key”? Handle the case when object keys are already column indices (e.g., when keys: []).
+
+					// FIXME: What if objects have “new” keys (i.e., index is not a number), e.g., the user wants to add new columns with data?
+					// What should we do? Should we add them at the end of the corresponding row? Will ret.push(obj[key]) do?
```

Comment:
What currently happens? We should definitely not *lose* data. Adding them to the end seems fine.

Reply:
I'm afraid, currently, nothing is happening. So we have another bug that should be fixed. I'll add new data to the end in the “order of appearance” inside of an object then.